doc/porting_guide.md

```diff
@@ -3,268 +3,292 @@
 
 _Default_ values will be chosen for you if you do not set the TARGET or CONFIG makefile variables.
 
-The TARGET and CONFIG flags are enumerated below, and can be mixed and matched in any order and are separated by the - (minus sign / dash) character.
+The TARGET and CONFIG flags are enumerated below, and can be mixed and matched in any order.  Each flag is separated by the - (minus sign / dash) character.
 
-##### TARGET flags
+##### TARGET Flags
 
 A typical TARGET flag looks like this:
 
     osx-static-debug
 
-###### Platform flag
+###### Platform Flag
 
     - [ osx | linux | arm-linux ]   - selects the target platform
 
-###### Output Library type
+###### Output Library Type
 
     - [ static | dynamic ]          - selects the output library type
 
 ###### Build Type
 
     - [ debug | release ]           - selects the build type
 
-##### CONFIG flags
+##### CONFIG Flags
 
 A typical CONFIG flag:
 
-    posix_io-posix_fs-thread_module-posix_platform-tls-senml-control_topic-memory_limiter
+    posix_io-posix_fs-posix_platform-tls-senml-memory_limiter
 
 ###### Xively Client Feature flags
 
     - control_topic     - the control topic feature is in development.
                           Currently activating this feature has no behavioral benefit.
     - senml             - turns on SENML JSON serialization support for timeseries data. To maintain a small
                           footprint size we recommend turning this off (by removing senml flag from CONFIG).
-    - threading         - turns on threading: application callbacks will be called on separate thread.
+    - threading         - POSIX only.  This turns causes application callbacks to be called on separate thread.
                           Not having this flag set application callbacks are called on the sole main
-                          thread of the Xively C Client. Only POSIX implementation is available.
+                          thread of the Xively C Client.
 
 ###### File System flags
 
     - posix_fs          - POSIX implementation of File System calls
     - memory_fs         - an in-memory File System implementation
     - dummy_fs          - empty implementation of File System calls
-
     - expose_fs         - adds a new Xively C Client API function which allows external definition
-                          of File System calls
+                          of File System calls.  This will be part of our BSP system in the future.
 
 ###### Development flags
 
-    - memory_limiter    - turns on memory limiting and monitoring functionality. The purpose of this configuration
-                          option is to aid development processes by simulating low-memory environment behavior.
+    - memory_limiter    - turns on memory limiting and monitoring functionality. The purpose of this configuration is to aid development processes by simulating low-memory environment behavior.
```

Comment:
unfortunately not manually breaking these lines ends up in needing the reader to scroll horizontally. Breaking manually brings more convenient experience for the reader, so pls break those manually as animals break lines. sorry.